src/Microsoft.IdentityModel.Tokens/Validation/Results/Details/LifetimeValidationError.cs

```diff
@@ -50,33 +34,45 @@ internal override Exception GetException()
         {
             if (ExceptionType == typeof(SecurityTokenNoExpirationException))
             {
-                return new SecurityTokenNoExpirationException(MessageDetail.Message);
+                var exception = new SecurityTokenNoExpirationException(MessageDetail.Message, InnerException);
```

Comment:
Would we want to make another constructor that takes a ValidationError rather than setting it in a separate step?

Reply:
I'll explore this possibility in the near future. Thanks for the input.